shell.go

```diff
@@ -72,16 +73,39 @@ func NewShell(url string) *Shell {
 	return NewShellWithClient(url, c)
 }
 
-func NewShellWithClient(url string, c *gohttp.Client) *Shell {
-	if a, err := ma.NewMultiaddr(url); err == nil {
-		_, host, err := manet.DialArgs(a)
-		if err == nil {
-			url = host
+func NewShellWithClient(url string, client *gohttp.Client) *Shell {
+	var sh Shell
+	sh.httpcli = *client
+
+	if maddr, err := ma.NewMultiaddr(url); err == nil {
+		if network, host, err := manet.DialArgs(maddr); err == nil {
+			if network == "unix" {
+				url = network
+				if tpt, ok := sh.httpcli.Transport.(*gohttp.Transport); ok {
+					var tptCopy *gohttp.Transport
+					if tpt == nil {
+						tptCopy = &gohttp.Transport{
+							Proxy:             gohttp.ProxyFromEnvironment,
+							DisableKeepAlives: true,
+						}
+					} else {
+						tptCopy = tpt.Clone()
```

Comment:
We don't _really_ need to _deep_ copy it but that is safer. However, if we do so, we need to bump the minimum go version in CI (.travis.yml) (it's failing).

Reply:
Done!